### PR TITLE
after restart kubelet, check kubelet status until available and wait until it is stable

### PR DESCRIPTION
**Pull request by pacoxu**
test on #107062

### CODE REVIEW

File: test/e2e_node/util.go [hunk 1]
```diff
@@ -408,6 +408,9 @@ func restartKubelet(running bool) {
 
 	stdout, err = exec.Command("sudo", "systemctl", "restart", kubeletServiceName).CombinedOutput()
 	framework.ExpectNoError(err, "Failed to restart kubelet with systemctl: %v, %s", err, string(stdout))
+
+	// testing, some test failed after kubelet restarts.
+	time.Sleep(5 * time.Second)
```
Comment (aojea) [inline test/e2e_node/util.go:413]: and a wait loop checking for `systemct status kubelet`?

Comment (pacoxu) [reply test/e2e_node/util.go]: good idea.

Some special cases: However, what if kubelet crashed after it started quickly. The check may pass but kubelet crashes then. (If so, the test should fail.)

```
_output/local/go/src/k8s.io/kubernetes/test/e2e_node/resource_usage_test.go:86
Unexpected error:
    <*errors.errorString | 0xc001746000>: {
        s: "No resource usage data for kubelet container (/kubelet.slice)",
    }
    No resource usage data for kubelet container (/kubelet.slice)
occurred
_output/local/go/src/k8s.io/kubernetes/test/e2e_node/resource_usage_test.go:194
```

Comment (aojea) [reply test/e2e_node/util.go]: just maintain the condition for X seconds ?
wait until `systemctl status is ok during 3 seconds`

File: test/e2e_node/util.go [hunk 2]
```diff
@@ -408,6 +408,13 @@ func restartKubelet(running bool) {
 
 	stdout, err = exec.Command("sudo", "systemctl", "restart", kubeletServiceName).CombinedOutput()
 	framework.ExpectNoError(err, "Failed to restart kubelet with systemctl: %v, %s", err, string(stdout))
+
+	// wait until the kubelet status succeed
+	gomega.Eventually(func() bool {
+		_, err = exec.Command("sudo", "systemctl", "status", kubeletServiceName).CombinedOutput()
+		return err == nil
+	}, 30*time.Second, 1*time.Second).Should(gomega.BeTrue())
+
```
Comment (aojea) [inline test/e2e_node/util.go:417]: and here something like
```
    gomega.Consistently(func() bool {
        	_, err = exec.Command("sudo", "systemctl", "status", kubeletServiceName).CombinedOutput()
		return err == nil
    }, 10*time.Second, 2*time.Second)).Should(gomega.BeTrue()))
```

File: test/e2e_node/util.go [hunk 3]
```diff
@@ -408,6 +408,13 @@ func restartKubelet(running bool) {
 
 	stdout, err = exec.Command("sudo", "systemctl", "restart", kubeletServiceName).CombinedOutput()
 	framework.ExpectNoError(err, "Failed to restart kubelet with systemctl: %v, %s", err, string(stdout))
+
+	// wait until the kubelet status is stable
```
Comment (aojea) [inline test/e2e_node/util.go:412]: I think you need the `Eventually` first, so
- wait until kubelet is ready
- once is ready, check it is always ready

Comment (pacoxu) [reply test/e2e_node/util.go:418]: Misunderstand that. 

Not sure. `If it can run consistently for 10s, I thought it is enough.`

Comment (aojea) [reply test/e2e_node/util.go:418]: I've made up the numbers, don't know what values are the best here honestly

**Comment by pacoxu** [comment]
> /retest
> 
> are the failure real? or flake?

A real failure. Bobby is trying to fix it. https://github.com/kubernetes/kubernetes/issues/107062#issuecomment-1009446893

However, this change would help to make things clear. If kubelet restarts and crashes quickly, we would find the problem directly(Or the failure would make use confused).

File: test/e2e_node/util.go [hunk 4]
```diff
@@ -408,6 +408,19 @@ func restartKubelet(running bool) {
 
 	stdout, err = exec.Command("sudo", "systemctl", "restart", kubeletServiceName).CombinedOutput()
 	framework.ExpectNoError(err, "Failed to restart kubelet with systemctl: %v, %s", err, string(stdout))
+
+	// wait until the kubelet status succeed
+	gomega.Eventually(func() bool {
+		_, err = exec.Command("sudo", "systemctl", "status", kubeletServiceName).CombinedOutput()
```
Comment (ruiwen-zhao) [inline test/e2e_node/util.go:414]: IIUC about the code [1][2], `CombinedOutput` does not return error as long as the command runs, even if stderror is not empty.  Should we use something like `sudo systemctl is-active kubelet` and assert on the output?

[1] https://cs.opensource.google/go/go/+/refs/tags/go1.17.6:src/os/exec/exec.go;l=557
[2] https://cs.opensource.google/go/go/+/refs/tags/go1.17.6:src/os/exec/exec.go;drc=refs%2Ftags%2Fgo1.17.6;l=337

**Pull request closed without merging**